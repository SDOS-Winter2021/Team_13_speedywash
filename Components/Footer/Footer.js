/**
 * @module
 */
import React from 'react'
import { Text, View, StyleSheet, Button, TouchableOpacity } from 'react-native'
import { FontAwesome5, MaterialIcons, AntDesign } from '@expo/vector-icons';
import keys from "../../configs/KEYS"

/**
 * Code for footer is present here
 * Three icons clicking on any of them updates the currentView.screen
 * which in return updates whole parent view
 * @param {Object} obj - An object
 * @param {module:globaltypes.VisibleView} obj.currentView - current visible/selected view
 * @param {function} obj.setcurrentView - current view setter
*/
function Footer({ currentView, setcurrentView }) {
    return (
        <View style={styles.footerStyle}>

            <TouchableOpacity style={styles.icons} onPress={()=>setcurrentView({screen: keys.screens.HOME, header: true, footer: true})} >
                <FontAwesome5 name="home" size={32} color={currentView.screen==keys.screens.HOME ? keys.colors.FOOTER_SELECTED : keys.colors.FOOTER_NORMAL} />
            </TouchableOpacity>

            <TouchableOpacity style={styles.icons} onPress={()=>setcurrentView({screen: keys.screens.PROFILE, header: true, footer: true})}>
                <FontAwesome5 name="user-circle" size={32} color={currentView.screen==keys.screens.PROFILE ? keys.colors.FOOTER_SELECTED : keys.colors.FOOTER_NORMAL}  />
            </TouchableOpacity>
        </View>
    )
}

const styles = StyleSheet.create({
    footerStyle: {
        // flex: 1,
        height: '100%',
        flexDirection: "row",
        backgroundColor: keys.colors.MAIN,
        justifyContent: 'space-around'
    },
    icons: {
        alignSelf: 'center'
    }
})

export default Footer;