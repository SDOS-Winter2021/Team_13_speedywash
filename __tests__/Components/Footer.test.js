import React from 'react';
import renderer from 'react-test-renderer';
import Footer from "../../Components/Footer/Footer"
import {test_user} from "../global"

it('Footer renders correctly', () => {
    const tree = renderer.create(<Footer currentView={{screen: "HOMEPAGE"}} />).toJSON();
    expect(tree).toMatchSnapshot();
})

