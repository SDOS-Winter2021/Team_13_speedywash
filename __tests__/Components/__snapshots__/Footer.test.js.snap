// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#006BAD",
      "flexDirection": "row",
      "height": "100%",
      "justifyContent": "space-around",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "opacity": 1,
      }
    }
  >
    <Text />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "opacity": 1,
      }
    }
  >
    <Text />
  </View>
</View>
`;
